lisa/sut_orchestrator/baremetal/cluster/idrac.py

```diff
@@ -50,14 +56,47 @@ def _start(self, wait: bool = True) -> None:
             self._log.debug("System is already powered on.")
             return
         self.cluster.reset("On")
+        check_till_timeout(
+            lambda: self.cluster.get_power_state() == "On",
+            timeout_message="wait for client into 'On' state",
+        )
         self._logout()
 
     def _restart(self, wait: bool = True) -> None:
         self._login()
         self.cluster.reset("ForceRestart")
+        check_till_timeout(
+            lambda: self.cluster.get_power_state() == "On",
+            timeout_message="wait for client into 'On' state",
+        )
         self._logout()
 
 
+class IdracSerialConsole(features.SerialConsole):
+    def _login(self) -> None:
+        platform: BareMetalPlatform = self._platform  # type: ignore
+        self.cluster: Idrac = platform.cluster  # type: ignore
+        self.cluster.login()
+
+    def _logout(self) -> None:
+        platform: BareMetalPlatform = self._platform  # type: ignore
+        self.cluster = platform.cluster  # type: ignore
+        self.cluster.logout()
+
+    def _get_console_log(self, saved_path: Optional[Path]) -> bytes:
+        self._login()
+        if saved_path:
+            screenshot_file_name: str = "serial_console"
+            decoded_data = base64.b64decode(self.cluster.get_server_screen_shot())
+            screenshot_raw_name = saved_path / f"{screenshot_file_name}.png"
+            img_file = open(screenshot_raw_name, "wb")
```

Comment:
Use `with` make sure the handle is closed.

Reply:
update